test/parallel/test-http-agent-timeout.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+{
+  // Ensure reuse of successful sockets.
+
+  const agent = new http.Agent({ keepAlive: true });
+
+  const server = http.createServer((req, res) => {
+    res.end();
+  });
+
+  server.listen(0, common.mustCall(() => {
+    let socket;
+    http.get({ port: server.address().port, agent })
+      .on('response', common.mustCall((res) => {
+        socket = res.socket;
+        assert(socket);
+        res.resume();
+        socket.on('free', common.mustCall(() => {
+          http.get({ port: server.address().port, agent })
+            .on('response', common.mustCall((res) => {
+              assert.strictEqual(socket, res.socket);
+              assert(socket);
+              agent.destroy();
+              server.close();
+            }));
+        }));
+      }));
+  }));
```

Comment:
Nit: I think we can simplify this to:
```suggestion
    http.get({ port: server.address().port, agent })
      .on('response', common.mustCall((res) => {
        const socket = res.socket;
        assert(socket);
        res.resume();
        socket.on('free', common.mustCall(() => {
          http.get({ port: server.address().port, agent })
            .on('response', common.mustCall((res) => {
              assert.strictEqual(socket, res.socket);
              agent.destroy();
              server.close();
            }));
        }));
      }));
  }));
```

Reply:
(Multiline suggestion is kinda confusing - basically moves `socket` declaration to the first callback and removes `assert(socket)` in the second callback as it basically cannot fail)